input/postgres/vacuum_progress.go

```diff
@@ -13,7 +13,18 @@ SELECT (EXTRACT(epoch FROM a.query_start)::int::text || pg_catalog.to_char(pid,
 			 (EXTRACT(epoch FROM COALESCE(backend_start, pg_catalog.pg_postmaster_start_time()))::int::text || pg_catalog.to_char(pid, 'FM0000000'))::bigint AS backend_identity,
 			 a.datname,
 			 (SELECT pg_catalog.regexp_matches(query, 'autovacuum: VACUUM (ANALYZE )?([^\.]+).([^\(]+)( \(to prevent wraparound\))?'))[2] AS nspname,
-			 (SELECT pg_catalog.regexp_matches(query, 'autovacuum: VACUUM (ANALYZE )?([^\.]+).([^\(]+)( \(to prevent wraparound\))?'))[3] AS relname,
+			 CASE
+			   WHEN ($1 = '' OR
+				   (COALESCE(n.nspname,
+					  (SELECT pg_catalog.regexp_matches(query, 'autovacuum: VACUUM (ANALYZE )?([^\.]+).([^\(]+)( \(to prevent wraparound\))?'))[2])
+					  || '.' ||
+					  COALESCE(c.relname,
+					  (SELECT pg_catalog.regexp_matches(query, 'autovacuum: VACUUM (ANALYZE )?([^\.]+).([^\(]+)( \(to prevent wraparound\))?'))[3])) !~* $1)
+			   THEN COALESCE(c.relname,
+				   (SELECT pg_catalog.regexp_matches(query, 'autovacuum: VACUUM (ANALYZE )?([^\.]+).([^\(]+)( \(to prevent wraparound\))?'))[3])
+			   ELSE
+				   ''
+		   END AS relname,
```

Comment:
Unfortunately this one gets pretty gnarly. I can try rewriting with a CTE to avoid duplication, or we can cross our fingers that we don't have to touch this much. Thoughts?

Reply:
Hmm, yeah, this becomes very difficult to read due to the regexp. Doing this in a CTE would certainly read easier.

Since we are dealing with a low volume scenario (i.e. there is at most a handful of vacuums running), we could also do the filtering in the Go code, if we felt that was easier to do - after all, one motivation for picking regexp as a format was that its (mostly) portable between Postgres, Go and other languages.

Reply:
I think I'd like to stick to SQL where possible just for consistency. Let me try refactoring this.

Reply:
Refactored and I think this is much more readable. Good call :+1:.